### PR TITLE
refactor(*):api objects and function binding implementation

### PR DESCRIPTION
**Pull request by arteevraina**
I have introduced new variables to handle office JS API objects more clearly and improved the code to look better.
Please Review this :) 

Signed-off-by: Arteev Raina <arteevraina@gmail.com>

### CODE REVIEW

File: src/SmartClauseList.js [hunk 1]
```diff
@@ -22,15 +21,19 @@ class SmartClauseList extends React.Component {
       items: [],
       message: ''
     }
+    this.bind  = this.getBindings.bind(this);
```
Comment (dselman) [inline src/SmartClauseList.js]: I don't understand why this is assigned 3 times.

**Review by dselman — CHANGES_REQUESTED** [review]

File: src/NewSmartClauseDialog.js [hunk 2]
```diff
@@ -55,7 +55,9 @@ class NewSmartClauseDialog extends React.Component {
   handleOk = () => {
     this.setState({ open: false });
     const that = this;
-    window.Office.context.document.bindings.addFromSelectionAsync(window.Office.BindingType.Text, { id: that.state.clauseId + '/' + that.state.templateId }, function (asyncResult) {
+    const Office = window.Office;
+    const bindings = window.Office.context.document.bindings;
```
Comment (DianaLease) [inline src/NewSmartClauseDialog.js]: can't this now be `const bindings = Office.context.document.bindings;`?

File: src/SmartClauseList.js (hunk 1 above)
Comment (DianaLease) [reply src/SmartClauseList.js]: You shouldn't need to bing these methods at all in the constructor since you are using arrow functions to define them.

**Review by DianaLease — CHANGES_REQUESTED** [review]

File: src/NewSmartClauseDialog.js (hunk 2 above)
Comment (arteevraina) [reply src/NewSmartClauseDialog.js]: Yes @DianaLease I will correct this.

File: src/SmartClauseList.js (hunk 1 above)
Comment (arteevraina) [reply src/SmartClauseList.js]: @DianaLease if i will not bind these methods in constructor i will get error saying that particular function is undefined.

Comment (arteevraina) [reply src/SmartClauseList.js]: @dselman I made a typo by binding method `gotoBinding` twice. I will fix that in my upcoming commit.

Comment (DianaLease) [reply src/SmartClauseList.js]: @arteevraina your refactor that uses the arrow functions means that they will get bound automatically so you can use `this.getBindings` or `this. gotoBinding` without binding them in the constructor. You shouldn't need to use `this.bind`.

Comment (DianaLease) [reply src/SmartClauseList.js]: so for example, where you use `this.gotoBinding.bind(item.id)` in the code can change to just `this.gotoBinding(item.id)`

**Review by DianaLease — CHANGES_REQUESTED** [review]
Your changes look great, but it looks like some of the new changes that went in a a recent PR are now getting reverted in yours (ie. refactoring to destructure { Component } from react. Not sure if maybe a merge got messed up somehow? Can you double-check those changes?

**Comment by arteevraina** [comment]
Yes some of the changes in recent PR are now being reflected in mine. What can i do to fix it? @DianaLease

**Comment by DianaLease** [comment]
> Yes some of the changes in recent PR are now being reflected in mine. What can i do to fix it? @DianaLease

Is it possible to revert any changes you didn't make so that your PR is just the things you intended to change?

**Comment by arteevraina** [comment]
@DianaLease I think i can delete those files which i have not changed in this PR. But it will create a new commit on deleting each file that is not required.

**Pull request closed without merging**